### PR TITLE
Health Checks Section

### PR DESCRIPTION
**Pull request by Poltergeist**
healthchecks mov

This introduces the health check section.

notes:
 - Endpoints is not introduced yet since ports is not available yet.

### CODE REVIEW

**Comment by philipnrmn** [comment]
Looks great! Nice work :)

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+            <FieldLabel>Max Failures</FieldLabel>
+            <FieldInput
+              name={`healthChecks.${key}.maxConsecutiveFailures`}
+              type="number"
+              min="0"
+              value={healthCheck.maxConsecutiveFailures}/>
+            <FieldError>{errors.healthChecks[key]}</FieldError>
+          </FormGroup>
+        </div>
+      </CollapsibleContainer>);
```
Comment (mlunoe) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: #nit let's get that paren on next line

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+  getHTTPFields(healthCheck, key) {
+    const {errors} = this.props;
+
+    if (healthCheck.protocol !== 'HTTP' && healthCheck.protocol !== 'HTTPS') {
+      return null;
+    }
+
+    return [
+      (<div className="flex row" key="path">
```
Comment (mlunoe) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: #nit let's move the parentheses out to a separate line

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+            <FieldHelp>Morbi leo risus</FieldHelp>
+          </FieldLabel>
+          <FieldError>{errors.healthChecks[key]}</FieldError>
+        </FormGroup>
+      </div>)
+    ];
+  }
+
+  getHealthChecksLines(data) {
+    const {errors} = this.props;
```
Comment (mlunoe) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: Why not just `const {healthChecks} = this.props.errors`, since that's the only one being used?

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+      (<div className="row flex" key="HTTPS">
+        <FormGroup showError={false} className="column-12">
+          <FieldLabel>
+            <FieldInput
+              checked={healthCheck.protocol === 'HTTPS'}
+              name={`healthChecks.${key}.https`}
+              type="checkbox"
+              value="HTTPS"/>
+            Make HTTPS
+            <FieldHelp>Morbi leo risus</FieldHelp>
```
Comment (mlunoe) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: Let's remove this

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+            type="text"
+            value={healthCheck.command}/>
+          <FieldError>{errors.healthChecks[key]}</FieldError>
+        </FormGroup>
+      </div>
+    );
+  }
+
+  getHTTPFields(healthCheck, key) {
+    const {errors} = this.props;
```
Comment (mlunoe) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: Why not just const {healthChecks} = this.props.errors, since that's the only one being used?

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+          </div>
+        </div>
+      );
+    });
+  }
+
+  render() {
+    let {data} = this.props;
+
+    // data = {healthChecks: [{}, {type: 'COMMAND'}]};
```
Comment (mlunoe) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: Let's remove this

File: src/styles/components/form-elements/styles.less [hunk 7]
```diff
@@ -34,4 +34,10 @@
     left: 100%;
     position: absolute;
   }
+
+  .form-remove {
+    position: absolute;
```
Comment (mlunoe) [inline src/styles/components/form-elements/styles.less]: Why is this necessary?

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+
+    if (healthCheck.protocol !== 'HTTP' && healthCheck.protocol !== 'HTTPS') {
+      return null;
+    }
+
+    return [
+      (<div className="flex row" key="path">
+        <FormGroup
+          className="column-6"
+          required={false}
```
Comment (mlunoe) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: No need to add `required` if not `true`

File: src/styles/components/form-elements/styles.less (hunk 7 above)
Comment (jfurrow) [inline src/styles/components/form-elements/styles.less]: @mlunoe This is to position the "remove" icon
 and I do think absolute positioning makes sense here, personally.

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+    ];
+  }
+
+  getHealthChecksLines(data) {
+    const {errors} = this.props;
+
+    return data.map((healthCheck, key) => {
+      return (
+        <div key={key} className="panel pod-short">
+          <div className="pod-narrow pod-short">
```
Comment (jfurrow) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: I think we should create a simple component for this. Something like `FormGroupContainer` or something to indicate that it is actually a group of form fields. Then maybe it can have a prop like `removable`, and when `removable={true}`, the component displays the `form-remove` element (of course it would have to accept an `onRemove` prop too). I think this would help us be more consistent. What do you think?

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js (hunk 1 above)
Comment (Poltergeist) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: done.

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js (hunk 5 above)
Comment (Poltergeist) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: done.

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js (hunk 2 above)
Comment (Poltergeist) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: done.

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js (hunk 8 above)
Comment (Poltergeist) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: done.

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js (hunk 4 above)
Comment (Poltergeist) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: done.

File: plugins/services/src/js/components/forms/HealthChecksFormSection.js (hunk 6 above)
Comment (Poltergeist) [inline plugins/services/src/js/components/forms/HealthChecksFormSection.js]: done.

**Comment by jfurrow** [comment]
Great work! 🚢  @mlunoe what do you think?

**Comment by Poltergeist** [comment]
This is now pointing to #1369

File: src/styles/components/form-elements/styles.less (hunk 7 above)
Comment (mlunoe) [inline src/styles/components/form-elements/styles.less]: Arh, I see
